Microsoft.Toolkit.Uwp.Services/Services/MicrosoftTranslator/TranslatorServiceClient.cs

```diff
@@ -0,0 +1,284 @@
+        /// <remarks><para>This method perform a non-blocking request for language code.</para>
+        /// <para>For more information, go to http://msdn.microsoft.com/en-us/library/ff512427.aspx.
+        /// </para></remarks>
+        /// <seealso cref="GetLanguagesAsync"/>
+        /// <seealso cref="Language"/>
+        public async Task<string> DetectLanguageAsync(string text)
+        {
+            if (string.IsNullOrWhiteSpace(text))
+            {
+                throw new ArgumentNullException(nameof(text));
+            }
+
+            text = text.Substring(0, Math.Min(text.Length, _MaxTextLengthForAutoDetection));
+
+            // Checks if it is necessary to obtain/update access token.
+            await this.CheckUpdateTokenAsync().ConfigureAwait(false);
+
+            var uri = string.Format(_detectUri, Uri.EscapeDataString(text));
+            var content = await client.GetStringAsync(uri).ConfigureAwait(false);
+
+            var doc = XDocument.Parse(content);
+            var detectedLanguage = doc.Root.Value;
+
+            return detectedLanguage;
+        }
+
+        /// <summary>
+        /// Initializes the <see cref="TranslatorServiceClient"/> class by getting an access token for the service.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> that represents the initialize operation.</returns>
+        /// <remarks>Calling this method isn't mandatory, because the token is get/refreshed everytime is needed. However, it is called at startup, it can speed-up subsequest requests.</remarks>
+        public Task InitializeAsync() => this.CheckUpdateTokenAsync();
+
+        /// <summary>
+        /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
+        /// </summary>
+        public void Dispose()
+        {
+            authToken.Dispose();
+            client.Dispose();
+        }
+
+        private async Task CheckUpdateTokenAsync()
+        {
+            if (string.IsNullOrWhiteSpace(SubscriptionKey))
+            {
+                throw new ArgumentException("Invalid Subscription Key. Go to Azure Portal and sign up for Microsoft Translator: https://portal.azure.com/#create/Microsoft.CognitiveServices/apitype/TextTranslation");
+            }
+
+            try
+            {
+                var token = await authToken.GetAccessTokenAsync().ConfigureAwait(false);
+                if (token != authorizationHeaderValue)
+                {
+                    // Updates the access token.
+                    authorizationHeaderValue = token;
+                    var headers = client.DefaultRequestHeaders;
+
+                    if (headers.Contains(_AuthorizationUri))
+                    {
+                        headers.Remove(_AuthorizationUri);
+                    }
+
+                    headers.Add(_AuthorizationUri, authorizationHeaderValue);
+                }
+            }
+            catch
```

Comment:
Why swallowing the exception here, maybe the developer need to know whats wrong so he can fix it or he can report the exception back to the community to find a solution for the problem